src/DynamoCore/Library/TypedParameter.cs

```diff
@@ -5,6 +5,19 @@
 
 namespace Dynamo.Library
 {
+    public struct TypedParameterParams
+    {
+        public TypedParameterParams(string parameterName, ProtoCore.Type type) : 
+            this()
+        {
+            ParameterName = parameterName;
+            Type = type;
+        }
+
+        public string ParameterName { get; set; }
+        public ProtoCore.Type Type { get; set; }
+        public object DefaultValue { get; set; }
+    }
```

Comment:
I think it is fine for a constructor to have multiple parameters. Now we need to construct `TypedParameterParams` which in most cases use same parameter of original `TypedParameter`.